### PR TITLE
Type annotations

### LINKED ISSUE #125: Type annotation for Python based Algorithms
## Description

Most of the python-based algorithms are not annotated by types as specified by [PEP 483](https://www.python.org/dev/peps/pep-0483/) and [PEP 484](https://www.python.org/dev/peps/pep-0484/) 

Using generic types to provide these improvements is an effective way to enforce good practices and provide general documentation for other users quickly.

## Implementation

This can be discussed using central generic types with specifically defined types for only those algorithms that work on some certain types (like counting sort) or each file having its independent types

### PR DESCRIPTION
**Pull request by UbadahJ**
Fixes #125 

@diptangsu Please review

### CODE REVIEW

File: Python/AlgorithmComparisons.py [hunk 1]
```diff
@@ -15,19 +15,19 @@
...
     print(heading)
     print('-' * len(heading))
     for algo, time_taken in comparison.items():
         print(f'{algo:<{largest_name_len}}{time_taken}')
 
 
-def compare_algorithms(numbers):
-    time_taken = {}
+def compare_algorithms(numbers) -> None:
+    time_taken: Dict[str, float] = {}
```
Comment (diptangsu) [inline Python/AlgorithmComparisons.py:30]: this is good

File: Python/AlgorithmComparisons.py [hunk 2]
```diff
@@ -39,14 +39,15 @@ def compare_algorithms(numbers):
     print_comparison(time_taken_sorted)
 
 
-print('COMPARISON OF SORTING ALGORITHMS (fastest to slowest):')
-print('------------------------------------------------------')
-print(f'Length of list: {len(random_numbers)}')
-print('\nElements randomly distributed:')
-compare_algorithms(random_numbers)
+if __name__ == '__main__':
```
Comment (diptangsu) [inline Python/AlgorithmComparisons.py:42]: good

File: Python/AlgorithmComparisons.py [hunk 3]
```diff
@@ -15,19 +15,19 @@
 random_numbers_reverse_sorted = sorted(random_numbers, reverse=True)
 
 
-def print_comparison(comparison):
-    largest_name_len = len(max(comparison.keys(), key=len))
+def print_comparison(comparison) -> None:
+    largest_name_len: int = len(max(comparison.keys(), key=len))
```
Comment (diptangsu) [inline Python/AlgorithmComparisons.py:19]: I think we can get rid of type annotations where the variable name is more than enough to make the reader understand the type. Let's remove the type annotations in places where it is not required.

Comment (UbadahJ) [reply Python/AlgorithmComparisons.py:19]: This is not intended for the reader but instead the static type checker. This makes sure the variable type is not changed whenever it gets reassigned. 

Try using PyRight on it

Comment (diptangsu) [reply Python/AlgorithmComparisons.py:19]: Okay cool then.

**Comment by diptangsu** [comment]
I had done a few changes before going to bed. Can you please pull and resolve the conflicts? Sorry about this haha..

**Pull request closed without merging**